src/Data/Primitive/Contiguous/Class.hs

```diff
@@ -569,7 +569,12 @@ class (Contiguous arr) => ContiguousU arr where
   -- | The unifted version of the mutable array type (i.e. eliminates an indirection through a thunk).
   type UnliftedMut arr = (r :: Type -> Type -> TYPE UnliftedRep) | r -> arr
 
-  -- | Resize an array into one with the given size.
+  -- | Resize an array into one with the given size. If the array is grown,
+  -- then reading from any newly introduced element before writing to it is undefined behavior.
+  -- The current behavior is that anything backed by @MutableByteArray#@ ends
+  -- uninitialized memory at these indices. But for @SmallMutableArray@, these
+  -- are set to an error thunk, so reading from them and forcing the result
+  -- causes the program to crash.
```

Comment:
```suggestion
  -- causes the program to crash. For @UnliftedArray@, the new elements have undefined values of an unknown type.
```

Reply:
I might just skip these lines 574-577, for instance the UnliftedArray (and PrimArray) cases aren't even type-safe.